sklearn/inspection/_partial_dependence.py

```diff
@@ -238,57 +259,103 @@ def partial_dependence(estimator, X, features, *, response_method='auto',
           :class:`~sklearn.ensemble.GradientBoostingRegressor`,
           :class:`~sklearn.ensemble.HistGradientBoostingClassifier`,
           :class:`~sklearn.ensemble.HistGradientBoostingRegressor`,
           :class:`~sklearn.tree.DecisionTreeRegressor`,
           :class:`~sklearn.ensemble.RandomForestRegressor`,
-          )
-          but is more efficient in terms of speed.
+          ) when `kind='average'`.
+          This is more efficient in terms of speed.
           With this method, the target response of a
           classifier is always the decision function, not the predicted
-          probabilities.
+          probabilities. Since the `'recursion'` method implicitely computes
+          the average of the Individual Conditional Expectation (ICE) by
+          design, it is not compatible with ICE and thus `kind` must be
+          `'average'`.
 
-        - 'brute' is supported for any estimator, but is more
+        - `'brute'` is supported for any estimator, but is more
           computationally intensive.
 
-        - 'auto': the 'recursion' is used for estimators that support it,
-          and 'brute' is used otherwise.
+        - `'auto'`: the `'recursion'` is used for estimators that support it,
+          and `'brute'` is used otherwise.
 
         Please see :ref:`this note <pdp_method_differences>` for
-        differences between the 'brute' and 'recursion' method.
+        differences between the `'brute'` and `'recursion'` method.
+
+    kind : {'legacy', 'average', 'individual', 'both'}, default='legacy'
+        Whether to return the partial dependence averaged across all the
+        samples in the dataset or one line per sample or both.
+        See Returns below.
+
+        Note that the fast `method='recursion'` option is only available for
+        `kind='average'`. Plotting individual dependencies requires using the
+        slower `method='brute'` option.
+
+        .. versionadded:: 0.24
+        .. deprecated:: 0.24
+            `kind='legacy'` is deprecated and will be removed in version 0.26.
+            `kind='average'` will be the new default. It is intended to migrate
+            from the ndarray output to :class:`~sklearn.utils.Bunch` output.
+
 
     Returns
     -------
-    averaged_predictions : ndarray, \
-            shape (n_outputs, len(values[0]), len(values[1]), ...)
-        The predictions for all the points in the grid, averaged over all
-        samples in X (or over the training data if ``method`` is
-        'recursion'). ``n_outputs`` corresponds to the number of classes in
-        a multi-class setting, or to the number of tasks for multi-output
-        regression. For classical regression and binary classification
-        ``n_outputs==1``. ``n_values_feature_j`` corresponds to the size
-        ``values[j]``.
+    predictions : ndarray or :class:`~sklearn.utils.Bunch`
+
+        - ``kind='legacy'``, ndarray of shape (n_outputs, len(values[0]),
```

Comment:
I think this will not format very well in HTML (see [rendering](https://104597-843222-gh.circle-artifacts.com/0/doc/modules/generated/sklearn.inspection.partial_dependence.html)). I also think more words, like "If kind='legacy', return value is ..." wouldn't hurt.

Reply:
Done